commons/src/main/scala/internals/StatefulProcessor.scala

```diff
@@ -0,0 +1,83 @@
+package com.mfglabs.stream
+
+import akka.stream.stage._
+import akka.stream.{Attributes, FlowShape, Inlet, Outlet}
+
+/**
+ * Inspired by https://doc.akka.io/docs/akka/2.5/stream/stream-cookbook.html#working-with-io
+ */
+class StatefulProcessor[A, B, C](
+  zero  : => B,
+  f     : (B, A) => (Option[B], IndexedSeq[C]),
+  lastPushIfUpstreamEnds: B => IndexedSeq[C]
+) extends GraphStage[FlowShape[A, C]] {
+
+  val in = Inlet[A]("StatefulProcessor.in")
+  val out = Outlet[C]("StatefulProcessor.out")
+
+  override val shape = FlowShape.of(in, out)
+
+  override def createLogic(inheritedAttributes: Attributes): GraphStageLogic = new GraphStageLogic(shape) {
+    private var state: B  = _
+    private var buffer    = Vector.empty[C]
+    private var finishing = false
+
+    setHandler(out, new OutHandler {
+      override def onPull(): Unit = {
+        if( state == null ) state = zero // to keep the laziness of zero
```

Comment:
Doesn't look like the `if( <exp> )` whitespace matches the rest of this codebase...